components/quick_switch_modal/quick_switch_modal.tsx

```diff
@@ -17,33 +17,55 @@ import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
 import SuggestionList from 'components/suggestion/suggestion_list.jsx';
 import SwitchChannelProvider from 'components/suggestion/switch_channel_provider.jsx';
 import SwitchTeamProvider from 'components/suggestion/switch_team_provider.jsx';
-import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator.tsx';
+import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator';
 
 import {NoResultsVariant} from 'components/no_results_indicator/types';
 
 const CHANNEL_MODE = 'channel';
 const TEAM_MODE = 'team';
 
-export default class QuickSwitchModal extends React.PureComponent {
-    static propTypes = {
+type SwitchToChannelResult = {
+    data?: true;
+    error?: true;
+}
+type ProviderSuggestions = {
+    matchedPretext: any,
+    terms: string[],
+    items: any[],
+    component: SwitchChannelProvider|SwitchTeamProvider,
+}
 
-        /**
-         * The function called to hide the modal
-         */
-        onHide: PropTypes.func.isRequired,
+export type Props = {
 
-        /**
-         * Set to show team switcher
-         */
-        showTeamSwitcher: PropTypes.bool,
+    /**
+     * The function called to hide the modal
+     */
+    onHide: () => void;
 
-        actions: PropTypes.shape({
-            joinChannelById: PropTypes.func.isRequired,
-            switchToChannel: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+    /**
+     * Set to show team switcher
+     */
+    showTeamSwitcher: boolean;
+    actions: {
+        joinChannelById: (channelId: string) => Promise<any>;
+        switchToChannel: (channel: Channel) => Promise<SwitchToChannelResult>;
```

Comment:
We actually have a type for this already: ActionResult from `mattermost-redux/types/actions`